src/main/java/com/example/onlinebookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,39 @@
+package com.example.onlinebookstore.repository.book;
+
+import com.example.onlinebookstore.dto.BookSearchParameters;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.repository.SpecificationBuilder;
+import com.example.onlinebookstore.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book, BookSearchParameters> {
+    private SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParameters searchParams) {
```

Comment:
OPTIONAL: Actually this method violates open-closed principle. It is possible to implement strategy here or think how it could be without numbers of if statements